crates/goose/src/agents/sub_recipe_execution_tool/sub_recipe_execute_task_tool.rs

```diff
@@ -2,35 +2,50 @@ use mcp_core::{tool::ToolAnnotations, Content, Tool, ToolError};
 use serde_json::Value;
 
 use crate::agents::{
-    sub_recipe_execution_tool::lib::execute_tasks, tool_execution::ToolCallResult,
+    sub_recipe_execution_tool::lib::execute_tasks,
+    sub_recipe_execution_tool::task_types::ExecutionMode, tool_execution::ToolCallResult,
 };
+use mcp_core::protocol::JsonRpcMessage;
+use tokio::sync::mpsc;
+use tokio_stream;
 
 pub const SUB_RECIPE_EXECUTE_TASK_TOOL_NAME: &str = "sub_recipe__execute_task";
 pub fn create_sub_recipe_execute_task_tool() -> Tool {
     Tool::new(
         SUB_RECIPE_EXECUTE_TASK_TOOL_NAME,
         "Only use this tool when you execute sub recipe task.
-EXECUTION STRATEGY:
-- DEFAULT: Execute tasks sequentially (one at a time) unless user explicitly requests parallel execution
-- PARALLEL: Only when user explicitly uses keywords like 'parallel', 'simultaneously', 'at the same time', 'concurrently'
+EXECUTION STRATEGY DECISION:
+1. PRE-CREATED TASKS: If tasks were created by subrecipe__create_task_* tools, check the execution_mode in the response:
+   - If execution_mode is 'parallel', use parallel execution
+   - If execution_mode is 'sequential', use sequential execution
+   - Always respect the execution_mode from task creation to maintain consistency
```

Comment:
IMO the model shouldn't pick how the executor runs a set of tasks. The model will always get the results back together, so it only impacts the implementation details of the tasks themselves. If the model decides it needs the result of one task to run the next, it will have to separate them anyway.